### PR TITLE
Speed up fab-panel create 2.8x

### PR DESCRIPTION
**Pull request by akhilles**
`fab-panel create` on a 5x 16MB assembly-panel mix: 50s -> ~18s, byte-identical output.

- The spatial copper-balance solve runs its per-layer passes on scoped threads instead of serially, with the density kernel stored CSR and per-layer scratch reused across iterations.
- The final document is no longer schema-validated at runtime (~20s on 100MB outputs, dominated by uppsala re-walking the DOM for each of IPC-2581C's ~30 descendant-axis keyrefs). The fab-panel tests keep schema coverage; the production path still proves the document parses.

---

> [!NOTE]
> **Medium Risk**
> Production fab-panel output is no longer schema-validated at runtime (tests retain coverage); parallel copper-balance iteration and early exit could theoretically affect numerical edge cases though the PR claims byte-identical output.
> 
> **Overview**
> Cuts **`fab-panel create`** time (~50s → ~18s on a large 5× assembly mix) while keeping byte-identical output.
> 
> **Spatial copper-balance solve** runs each layer’s error and radius-update passes on scoped threads instead of serially. The lattice density kernel is stored in **CSR** form with **`smooth_into` / `smooth_adjoint_into`** writing into per-layer scratch reused across iterations. The solve **stops early** when the max radius change drops below a void-quantization threshold, and **`project_box_sum`** uses fewer bisection steps (64 → 44).
> 
> **Fabrication panel XML** no longer runs full IPC-2581 **schema validation** on the production path (~20s on ~100MB documents); it still **parses** the reformatted XML. Fab-panel **tests** continue to call **`Ipc2581::validate`**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 68085f540046f64baa45022df176ed5312d8d188. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: crates/pcb-ipc2581-tools/src/commands/fab_panel/xml.rs [hunk 1]
```diff
@@ -254,8 +254,8 @@ pub(super) fn write_fab_panel_xml(
         edit::apply(provisional, edits)?
     };
     let xml = crate::utils::format::reformat_xml(&xml)?;
-    Ipc2581::validate(&xml)
-        .context("Generated IPC-2581 fabrication panel XML failed schema validation")?;
+    // Schema validation lives in the fab-panel tests; on documents this size
+    // it costs ~20s, so the production path only proves the document parses.
```
[the remark this answers is not in the document]
Comment (akhilles) [reply crates/pcb-ipc2581-tools/src/commands/fab_panel/xml.rs:258]: Intentional, per maintainer direction: runtime schema validation costs ~20s on 100MB panels (uppsala re-walks the DOM per descendant keyref) and was judged not worth it; a vendored-fork fix was prototyped and rejected as unmaintainable. Schema coverage stays in the fab-panel tests, which validate the generator itself. The PR description was already updated to say exactly this — the "validation still on" text quoted here is from an earlier revision.

[the remark this answers is not in the document]
Comment (akhilles) [reply crates/pcb-ipc2581-tools/src/commands/fab_panel/xml.rs:258]: Intentional: this repo does not changelog performance work (maintainer preference), and the removed validation is an internal guard — the emitted document is unchanged.
